web/src/main/angular/src/app/core/components/transaction-timeline-v2/transaction-timeline-container-v2.component.ts

```diff
@@ -0,0 +1,133 @@
+    @ViewChild(TransactionTimelineComponentV2, { static: true }) transactionTimelineComponent: TransactionTimelineComponentV2;
+
+    private unsubscribe = new Subject<void>();
+
+    keyIndex: any;
+    startTime: number;
+    endTime: number;
+    rowData: any;
+    barRatio: number;
+
+    constructor(
+        private storeHelperService: StoreHelperService,
+        private transactionSearchInteractionService: TransactionSearchInteractionService,
+        private messageQueueService: MessageQueueService,
+        private analyticsService: AnalyticsService,
+        private cd: ChangeDetectorRef
+    ) {}
+
+    ngOnInit() {
+        this.connectStore();
+    }
+
+    ngOnDestroy() {
+        this.unsubscribe.next();
+        this.unsubscribe.complete();
+    }
+
+    private connectStore(): void {
+        this.storeHelperService.getTransactionDetailData(this.unsubscribe).pipe(
+            filter((transactionDetailInfo: any) => {
+                return transactionDetailInfo && transactionDetailInfo.transactionId ? true : false;
+            })
+        ).subscribe((transactionDetailInfo: ITransactionDetailData) => {
+            this.startTime = transactionDetailInfo.callStackStart;
+            this.endTime = transactionDetailInfo.callStackEnd;
+            this.keyIndex = transactionDetailInfo.callStackIndex;
+            this.barRatio = this.getBarRatio(transactionDetailInfo);
+            this.rowData = this.filterCallStack(transactionDetailInfo);
+            this.cd.detectChanges();
+        });
+    }
+
+    private getBarRatio(tInfo: ITransactionDetailData): number {
+        return 1000 / (tInfo.callStack[0][tInfo.callStackIndex.end] - tInfo.callStack[0][tInfo.callStackIndex.begin]);
+    }
+
+    private filterCallStack(rowData: ITransactionDetailData): any {
+        let data=[];
+        let async=[];
+        let asyncRootInfo = [[],[]];
+        let isAsync = false;
+
+        rowData.callStack.filter(call =>
+            (call[this.keyIndex.isMethod] && !call[this.keyIndex.excludeFromTimeline] && call[this.keyIndex.service] !== ''))
+            .forEach((call) => {
+                let depth = call[this.keyIndex.tab];
+                while (depth >= data.length) {
+                    data.push([]);
+                    async.push([]);
+                }
+
+                if(isAsync) {
+                    let asyncInfoLastIndex = asyncRootInfo[0].length - 1;
+                    if (call[this.keyIndex.applicationName] != asyncRootInfo[1][asyncInfoLastIndex]) {
+                        isAsync = false;
+                    } else if (call[this.keyIndex.tab] <= asyncRootInfo[0][asyncInfoLastIndex]) {
+                        asyncRootInfo[0].pop();
+                        asyncRootInfo[1].pop();
+                        isAsync = false;
+                    }
+                }
+
+                if(call[this.keyIndex.apiType] === "ASYNC") {
```

Comment:
Space after `if` statement
```suggestion
                if (call[this.keyIndex.apiType] === "ASYNC") {
```